### DISCUSSION TITLE
Possible to create jobs via OCAPI?

### DISCUSSION THREAD

**Discussion by jwilson-lyonscg** [SalesforceCommerceCloud/sfcc-ci#531] [Team Posts]
I see that you can execute predefined jobs and search for existing jobs using OCAPI. Are you able to do any sort of CRUD functionality of jobs? Create them? Delete them? etc? If not, is this something that is planned? A lot of opportunity of automate the management of multiple sandboxes if you had full control of creation of jobs via OCAPI.

**Comment by pravngaur** [comment]
Hello Jake,

Yes you can do that by importing the jobs in sandboxes, say if you are
setting up new sandboxes for the team or if it is a part of new story,
change.

This is feasible using SFCC build-suite, you can hook up with your CI
server to trigger the continuous builds or setup a job for the first time
sandbox setup.

**Comment by jwilson-lyonscg** [comment]
Thanks. How do you import jobs via OCAPI? Is this part of the Data API? Do you have a documentation link for this? I'm just not seeing it. Thank you again,

**Comment by wundrian** [comment]
Creating jobs isn't possible via OCAPI today, you can only trigger them. What pravngaur is referring to is the import functionality - you have to push a XML file with the job description, then use OCAPI to trigger an import of that job file.

**Comment by vivily** [comment]
There is a Data API that can create a job. Unfortunately, it's in Beta
version. Only internal Apps can use it.
com/demandware/wapi/resource/data/job/JobsIDResource.java

Wei

On Fri, Apr 5, 2019 at 6:50 PM Jake Wilson <notifications@github.com> wrote:

> I see that you can execute predefined jobs and search for existing jobs
> using OCAPI. Are you able to do any sort of CRUD functionality of jobs?
> Create them? Delete them? etc? If not, is this something that is planned? A
> lot of opportunity of automate the management of multiple sandboxes if you
> had full control of creation of jobs via OCAPI.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/19>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB6B-j9YGXV-9tEdbHms3Z3UPzW8nXA9ks5vd9MvgaJpZM4cf2Nb>
> .
>


-- 

<http://smart.salesforce.com/sig/wei.liu//us_mb/default/link.html>

**Comment by tobiaslohr** [comment]
@vivily Let's please focus only on APIs which are available today to partners and customers. The API you are referring to is not part of this.

**Comment by pravngaur** [comment]
Hello Jake,

Sorry could not reply back.

As of now, you can't do it with OCAPI, but with build suite only.

Request you to please explore the documentation....

**Comment by jwilson-lyonscg** [comment]
@wundrian Do you mean that if you push a properly formatted XML file to the instance via WebDAV that essentially "creates" a job?

**Comment by wundrian** [comment]
That properly formatted XML needs to be a jobs.xml, formatted according to
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/DWAPI/xsd/jobs.xsd
Package that in a site import zip file, upload it to your instance and
execute the sfcc-site-archive-import system job via OCAPI. Pass the zip
file as a parameter in merge mode. That should create the job(s) defined in
the xml file. You can then trigger them via OCAPI.

It's a bit of a dance but it works in our testing.

On Tue, Apr 9, 2019 at 7:11 PM Jake Wilson <notifications@github.com> wrote:

> @wundrian <https://github.com/wundrian> Do you mean that if you push a
> properly formatted XML file to the instance via WebDAV that essentially
> "creates" a job?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/19/comments/7>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABhi5TYR5pbHsxIi9-YdSwyXWT6OZie0ks5vfMmjgaJpZM4cf2Nb>
> .
>


-- 
Matti Bickel
Senior Performance Engineer
Leutragraben 2-4, 07743 Jena, Germany

**Comment by jwilson-lyonscg** [comment]
Great thanks. I've been going through the build suite tools understanding how it all comes together. Thank you,
